be/src/storage/rowset/column_writer.cpp

```diff
@@ -896,6 +898,7 @@ StringColumnWriter::StringColumnWriter(const ColumnWriterOptions& opts, std::uni
         : ColumnWriter(std::move(field), opts.meta->is_nullable()), _scalar_column_writer(std::move(column_writer)) {}
 
 Status StringColumnWriter::append(const vectorized::Column& column) {
+    DCHECK(check_string_lengths(column).ok());
```

Comment:
We can not use DCHECK here?
There is a small chance that the varchar's length will be out of bounds, if we use DCHECK, it will crash BE.

Reply:
We should let the error pass out and abort current write.